### ISSUE TITLE
fails to parse yaml block info without `not ok`

### ISSUE THREAD

**Issue by keithamus** [scottcorgan/tap-out#15]
Hi @scottcorgan! Thanks for the project, I've been playing with it recently - but I've noticed a couple of issues around yaml blocks I thought I'd address.
### Summary

If a test is marked as anything but `not ok` - but includes diagnostic output via a yaml block, then tap-out will crash, and not parse any of the results.
### Details

Here's the smallest TAP output I can generate that passes with a yaml block:

``` TAP
not ok 1 - a test
  ---
    a: b
  ...
```

Now, here's one that fails:

``` TAP
ok 1 - a test
  ---
    a: b
  ...
```

The error it fails with:

``` js
TypeError: Cannot read property 'name' of undefined
  at Parser._handleError (./tap-out/index.js:165:19)
  at Parser.handleLine (./tap-out/index.js:44:8)
  at Stream.<anonymous> (./tap-out/index.js:199:14)
  at emitOne (events.js:77:13)
  at Stream.emit (events.js:169:7)
  at drain (./through/index.js:36:16)
  at Stream.stream.queue.stream.push (./through/index.js:45:5)
  at emit (./tap-out/node_modules/split/index.js:37:14)
  at next (./tap-out/node_modules/split/index.js:49:7)
  at Stream.<anonymous> (./tap-out/node_modules/split/index.js:54:5)
```

It seems to be relying on there only being an error block when the test result is `not ok`. For example comments can be in between the test result and the yaml block and its fine:

``` TAP
not ok - a test
# Diagnostic
  ---
  a: b
  ...
```

Whereas if we just change `not ok` to `ok`, it fails with the same error above.

``` TAP
ok - a test
# Diagnostic
  ---
  a: b
  ...
```
### Other info
- You might be thinking "is this an error, why would an `ok` test have a diagnostic block?". Well, the [TAP 13 specification](http://testanything.org/tap-version-13-specification.html) does not explicitly mention that yaml blocks are only alowed for `not ok` tests. It also has a listing (the first one) that seems to suggest diagnostic reporting is acceptable for an `ok` block:  
  
  > TAP13's general format is:
  > 
  > ```
  > TAP version 13
  > 1..N
  > ok 1 Description # Directive
  > # Diagnostic
  > ---
  > message: 'Failure message'
  > severity: fail
  > data:
  >   got:
  >     - 1
  >     - 3
  >     - 2
  >   expect:
  >     - 1
  >     - 2
  >     - 3
  > ...
  > ok 47 Description
  > ok 48 Description
  > more tests....
  > ```
- I tried running the same tests through Substack's [`tap-parser`](https://github.com/substack/tap-parser), and, <strike>while it does not parse diagnostic yaml properly for `not ok`, it does not fail. I will also be filing an issue on `tap-parser` about this though.</strike> (don't know what I did wrong first time round, but I ran all of the above examples in tap-parser again and they're all fine).

**Comment by scottcorgan** [comment]
Will be fixed in the `2.0` rewrite.

**Comment by scottcorgan** [comment]
This should be fixed now
